sdk/src/main/java/io/opentelemetry/sdk/trace/SpanBuilderSdk.java

```diff
@@ -120,31 +146,112 @@
 
   @Override
   public Span.Builder setRecordEvents(boolean recordEvents) {
-    this.recordEvents = recordEvents;
     return this;
   }
 
   @Override
   public Span startSpan() {
-    // TODO: get remoteParent span from the context if noParent=false and remoteParent/remoteParents
-    // are null
-    // TODO: correctly implement this.
-    SpanContext context =
-        SpanContext.create(
-            TraceId.fromBytes(new byte[] {1, 2, 3, 4, 5, 6, 7, 8, 8, 7, 6, 5, 4, 3, 2, 1}, 0),
-            SpanId.fromBytes(new byte[] {1, 2, 3, 4, 5, 6, 7, 8}, 0),
-            TraceOptions.builder().setIsSampled(true).build(),
-            Tracestate.getDefault());
+    SpanContext parentContext = parent(parentType, parent, remoteParent);
+    SpanContext spanContext = createSpanContext(spanName, parentContext, sampler);
+    // TODO return DefaultSpan if not recording events
+    TimestampConverter timestampConverter = getTimestampConverter(parent);
     return RecordEventsReadableSpanImpl.startSpan(
-        context,
+        spanContext,
         spanName,
         spanKind,
-        null,
+        parentContext != null ? parentContext.getSpanId() : null,
         traceConfig,
         spanProcessor,
-        null,
-        MillisClock.getInstance(),
-        Resource.getEmpty());
+        timestampConverter,
+        clock,
+        resource);
+  }
+
+  private SpanContext createSpanContext(String name, SpanContext parentContext, Sampler sampler) {
+    TraceId traceId;
+    SpanId spanId = generateRandomSpanId(random);
+    Tracestate tracestate = Tracestate.getDefault();
+    if (parentContext == null || !parentContext.isValid()) {
+      // New root span.
+      traceId = generateRandomTraceId(random);
+      // This is a root span so no remote or local parent.
+    } else {
+      // New child span.
+      traceId = parentContext.getTraceId();
+      tracestate = parentContext.getTracestate();
+    }
+    Decision samplingDecision =
```

Comment:
In the result of the sampler Decision we do have some attributes which we need to add to the Span after creation.